spec/inferno/test_runner_spec.rb

```diff
@@ -33,34 +40,57 @@
       .to_return(status: 200)
   end
 
-  def error_results_message(error_results)
-    error_results.map do |r|
-      "#{r.runnable.title}: #{r.result_message}"
-    end.join("\n")
-  end
+  context 'without a bearer token' do
+    it 'runs' do
+      expect do
+        runner.run(group, { url: base_url, patient_id: patient_id })
+      end.to_not raise_error
+    end
 
-  it 'runs' do
-    expect do
-      runner.run(group, { url: base_url, patient_id: patient_id })
-    end.to_not raise_error
-  end
+    it 'creates results' do
+      test_count = group.tests.length
+
+      results = runner.run(group, { url: base_url, patient_id: patient_id })
 
-  it 'creates results' do
-    test_count = group.tests.length
+      expect(results.length).to eq(test_count + 1)
+    end
 
-    results = runner.run(group, { url: base_url, patient_id: patient_id })
+    it 'only contains no "error" results apart from the "error test"' do
+      results = runner.run(group, { url: base_url, patient_id: patient_id })
+      error_results =
+        results
+          .reject { |result| result.test&.title == 'error test' }
+          .reject { |result| result.test_group&.title == 'Demo Group Instance 1' }
+          .select { |result| result.result == 'error' }
 
-    expect(results.length).to eq(test_count + 1)
+      expect(error_results).to be_empty, error_results_message(error_results)
+    end
   end
 
-  it 'only contains no "error" results apart from the "error test"' do
-    results = runner.run(group, { url: base_url, patient_id: patient_id })
-    error_results =
-      results
-        .reject { |result| result.test&.title == 'error test' }
-        .reject { |result| result.test_group&.title == 'Demo Group Instance 1' }
-        .select { |result| result.result == 'error' }
+  context 'with a bearer token' do
```

Comment:
These tests don't actually check the new functionality unless we verify that the requests that are made include the bearer token.

Reply:
I reverted test_runner_spec and updated fhir_client_spec with new tests to check

- the contents of the security header
- the oauth2 and basic auth flags of the client